ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorsExecutor.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.execution;
+
+import io.confluent.ksql.KsqlExecutionContext;
+import io.confluent.ksql.parser.tree.ListConnectors;
+import io.confluent.ksql.parser.tree.ListConnectors.Scope;
+import io.confluent.ksql.rest.entity.ConnectorList;
+import io.confluent.ksql.rest.entity.ErrorEntity;
+import io.confluent.ksql.rest.entity.KsqlEntity;
+import io.confluent.ksql.rest.entity.KsqlWarning;
+import io.confluent.ksql.rest.entity.SimpleConnectorInfo;
+import io.confluent.ksql.services.ConnectClient;
+import io.confluent.ksql.services.ConnectClient.ConnectResponse;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.connect.runtime.ConnectorConfig;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorInfo;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorType;
+
+public final class ListConnectorsExecutor {
+
+  private ListConnectorsExecutor() { }
+
+  @SuppressWarnings("OptionalGetWithoutIsPresent")
+  public static Optional<KsqlEntity> execute(
+      final ConfiguredStatement<ListConnectors> configuredStatement,
+      final KsqlExecutionContext ksqlExecutionContext,
+      final ServiceContext serviceContext
+  ) {
+    final ConnectClient connectClient = serviceContext.getConnectClient();
+    final ConnectResponse<List<String>> connectors = serviceContext.getConnectClient().connectors();
```

Comment:
If I'm reading this correctly, `SHOW CONNECTORS` will list all connectors in the connect cluster, regardless of whether they were created through KSQL or not. Is this the intent? (Assume so, but want to check.) I see value both ways.

Reply:
yup this is correct